jvb-api/jvb-api-client/src/main/kotlin/org/jitsi/videobridge/api/util/XmppWebSocketClient.kt

```diff
@@ -121,4 +121,9 @@ class XmppWebSocketClient(
      * Connect the websocket client
      */
     fun run() = wsClient.run()
+
+    /**
+     * Disconnect the websocket client
+     */
+    fun stop() = wsClient.stop()
```

Comment:
Does this need to cancel the futures, also?  Otherwise won't `sendIqAndGetReply` block forever?

Reply:
There are timeouts there, so they won't block forever.  Also, I'm pretty sure I thought about this exact scenario last week and had a reason why I thought this would be sufficient, but it escapes me now, so I could go either way.  We could cancel the futures here and also set a sentinel to make sure any new requests returned immediately.

Reply:
I looked at doing this...I'd have to add some locking to make sure no new futures were added to the map when we were shutting down in addition to the sentinel (to make sure everything got cancelled), so I thought maybe it's not worth it...but let me know if you have thoughts.

Reply:
A timeout on `stop` is not great but is probably okay, if you've tested that it works.

Reply:
The fancier stop seems like it'd be a good idea, but maybe not essential if the timeout works...I don't imagine our normal JVB/Jicofo usecases will ever want to stop the JVB API code?

Reply:
I guess we'd stop the client when we ended a session with the bridge (because of how Jicofo is set up, we'll have a jvb api client per bridge per conference), but I don't think the timeout behavior feels like a big deal there.  I did add a test for stop and verified pending calls timeout.

Reply:
As long as a race condition doesn't cause jicofo in general to go into a timeout-block that blocks other transactions, I think this is fine.

Reply:
`stop` itself blocks on cancelling the job the websocket client is in, and anything waiting for a response is blocked on a future we created, so there shouldn't be any deadlock scenario there.